### DISCUSSION TITLE
How to dynamically change min and max of a slider

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#1149] [Q&A] [answered]
### Question

Currently we have to set min and max of a slider before `ui.run()`:
```py
slider = ui.slider(min=0, max=100, value=50)
```
However, there're apps we have to change the range of a slider dynamically. Taking this example: https://github.com/zauberzeug/nicegui/tree/main/examples/slideshow/, I would like users to provide a path to an image folder, then they can use a slider to walk through images (instead of using the keyboard). In this case, we need to set the max of the slider depending on the number of images in the folder. I tried a few workarounds such as writing a class wrapper or using the decorator @ui.refreshable. However they create a new slider instead of updating the created one.

**Comment by rodja** [comment]
It's working for me:

```py
def add_slider():
    ui.slider(min=0, max=random.randint(1, 100), value=100).props('label-always').classes('mb-4')

ui.button('add slider', on_click=add_slider)
```

Could you please provide a [minimal, executable code snippet reproducing the problem](https://en.wikipedia.org/wiki/Minimal_reproducible_example). This will allow us to identify and fix it more efficiently.

**Comment by falkoschindler** [comment, marked as answer]
Hi @nghia-vo!
I think you're looking for something like this:

```py
from nicegui import events, ui

def set_min(e: events.ValueChangeEventArguments):
    slider._props['min'] = e.value
    slider.update()

def set_max(e: events.ValueChangeEventArguments):
    slider._props['max'] = e.value
    slider.update()

min_input = ui.number('min', min=0, max=100, value=0, on_change=set_min)
max_input = ui.number('max', min=0, max=100, value=100, on_change=set_max)

slider = ui.slider(min=0, max=100, value=50).props('label-always')

ui.run()
```

**Comment by nghia-vo** [comment]
@falkoschindler : Thanks very much, that solution works. 
@rodja : As an example below, if I don't use Falko solution, the slider is created in the left drawer instead of being updated in the main page.

```python
from nicegui import events, ui

def set_min(e: events.ValueChangeEventArguments):
    slider._props['min'] = e.value
    slider.update()

def set_max(e: events.ValueChangeEventArguments):
    slider._props['max'] = e.value
    slider.update()

with ui.left_drawer().style('background-color: #d7e3f4'):
    with ui.row():
        min_input = ui.number('min', min=0, max=100, value=0,
                              on_change=set_min)
        max_input = ui.number('max', min=0, max=100, value=100,
                              on_change=set_max)

slider = ui.slider(min=0, max=100, value=50).props('label-always')
slice_pos = ui.label()
slice_pos.bind_text_from(slider, 'value')

ui.run()
```
